tests/sentry/api/endpoints/test_project_key_details.py

```diff
@@ -287,3 +311,20 @@ def test_simple(self):
         resp = self.client.delete(url)
         assert resp.status_code == 204, resp.content
         assert not ProjectKey.objects.filter(id=key.id).exists()
+
+    def test_use_case(self):
+        """Cannot delete an internal DSN"""
```

Comment:
What's the process if we need to delete/rotate an internal DSN?

Reply:
Super should be allowed to update / delete a key. Added a test for that now.